diff --git a/lua/project/util/history.lua b/lua/project/util/history.lua
--- a/lua/project/util/history.lua
+++ b/lua/project/util/history.lua
@@ -701,16 +701,6 @@ function M.write_history(path)
     file_history = #res > historysize and vim.list_slice(res, #res - historysize, #res) or res
   end
 
-  if #file_history == 0 then
-    vim.uv.fs_close(fd)
-    if vim.g.project_history_no_data_notified ~= 1 then
-      Log.error('(project.util.history.write_history): No data available to write!')
-      vim.notify('(project.util.history.write_history): No data available to write!', ERROR)
-      vim.g.project_history_no_data_notified = 1
-    end
-    return
-  end
-
   fd = path == Path.historyfile and M.open_history('w') or Path.open_file(path, 'w')
   if not fd then
     Log.error('(project.util.history.write_history): File restricted!')
